### PR TITLE
[CBRD-25486] Add test case for When removed the granted objects then set 'NULL' to grants column on the db_authorization table (11.4)

### PR DESCRIPTION
**Pull request by swi0110**
Refer to http://jira.cubrid.org/browse/CBRD-25486
It related http://jira.cubrid.org/browse/CBRD-25580

and It needs to backport release/10.2, release/11.2, release/11.3

### CODE REVIEW

File: sql/_35_fig_cake/cbrd_25486/answers/01_drop_table.answer [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+0
+===================================================
+grantor_name    grantee_name    object_type    object_name    owner_name    auth_type    is_grantable    
+U1     U2     CLASS     tbl     U1     SELECT     YES     
+U2     U3     CLASS     tbl     U1     SELECT     YES     
+
+===================================================
+owner.name    grants    
+DBA          
+U1
```
Comment (swi0110) [inline sql/_35_fig_cake/cbrd_25486/answers/01_drop_table.answer]: jdbc driver의 구현상의 문제로 db_authorization 테이블의 grants column은 출력되지 않습니다.

File: sql/_35_fig_cake/cbrd_25486/answers/01_drop_table.answer [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+null     
+
+===================================================
+0
+===================================================
+owner.name    grants    
+DBA          
+U1          
+U2          
+U3
```
Comment (swi0110) [inline sql/_35_fig_cake/cbrd_25486/answers/01_drop_table.answer]: jdbc driver의 구현상의 문제로 db_authorization 테이블의 grants column은 출력되지 않습니다.

File: sql/_35_fig_cake/cbrd_25486/answers/02_drop_view.answer [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+===================================================
+grantor_name    grantee_name    object_type    object_name    owner_name    auth_type    is_grantable    
+U1     U2     CLASS     tbl     U1     SELECT     YES     
+U1     U3     VCLASS     v1_tbl     U1     SELECT     YES     
+U2     U3     CLASS     tbl     U1     SELECT     YES     
+
+===================================================
+owner.name    grants    
+DBA          
+U1
```
Comment (swi0110) [inline sql/_35_fig_cake/cbrd_25486/answers/02_drop_view.answer]: jdbc driver의 구현상의 문제로 db_authorization 테이블의 grants column은 출력되지 않습니다.

File: sql/_35_fig_cake/cbrd_25486/answers/02_drop_view.answer [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+
+===================================================
+grantor_name    grantee_name    object_type    object_name    owner_name    auth_type    is_grantable    
+U1     U2     CLASS     tbl     U1     SELECT     YES     
+U2     U3     CLASS     tbl     U1     SELECT     YES     
+
+===================================================
+owner.name    grants    
+DBA          
+U1
```
Comment (swi0110) [inline sql/_35_fig_cake/cbrd_25486/answers/02_drop_view.answer]: jdbc driver의 구현상의 문제로 db_authorization 테이블의 grants column은 출력되지 않습니다.

File: sql/_35_fig_cake/cbrd_25486/cases/05_drop_partition_table.sql [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+drop table u1.t1_h;
+drop table u1.t1_r;
+
+
+evaluate 'connect to u1, drop u1.t1_h';
+call login(class db_user,'u1','');
+
+drop table u1.t1_h;
+
+select * from db_auth where grantee_name != 'PUBLIC';
```
Comment (ssihil) [inline sql/_35_fig_cake/cbrd_25486/cases/05_drop_partition_table.sql]:
```suggestion
select * from db_auth where grantee_name != 'PUBLIC' order by grantor_name, object_name;
```

**Review by ssihil — APPROVED** [review]

File: sql/_35_fig_cake/cbrd_25486/cases/05_drop_partition_table.sql (hunk 5 above)
Comment (swi0110) [reply sql/_35_fig_cake/cbrd_25486/cases/05_drop_partition_table.sql]: 반영 됐습니다.
review 감사합니다.

**Review by tw-kang — APPROVED** [review]

**Review by kwonhoil — APPROVED** [review]

File: sql/_35_fig_cake/cbrd_25486/answers/02_drop_view.answer (hunk 3 above)
Comment (bagus-kim) [reply sql/_35_fig_cake/cbrd_25486/answers/02_drop_view.answer]: @swi0110 문의사항입니다.
grants 가 NULL로 변경되는 것이 패치내역인데, 말씀하신 driver구현상 이슈로 비교를 못한다면, 다른 방법으로 비교가 되어야 하지 않나요?

File: sql/_35_fig_cake/cbrd_25486/cases/01_drop_table.sql [hunk 6]
```diff
@@ -0,0 +1,42 @@
+/* Verified the CBRD-25486
+As-is: if removed the object then remove the related data on the db_authorization table
+To-be: if removed the object then set 'NULL' relate data on the db_authorization table
```
Comment (bagus-kim) [inline sql/_35_fig_cake/cbrd_25486/cases/01_drop_table.sql]:
```suggestion
To-be: if deleted  the table then set 'NULL' relate data on the db_authorization table
```

File: sql/_35_fig_cake/cbrd_25486/cases/02_drop_view.sql [hunk 7]
```diff
@@ -0,0 +1,49 @@
+/* Verified the CBRD-25486
+As-is: if removed the object then remove the related data on the db_authorization table
+To-be: if removed the object then set 'NULL' relate data on the db_authorization table
```
Comment (bagus-kim) [inline sql/_35_fig_cake/cbrd_25486/cases/02_drop_view.sql]:
```suggestion
To-be: if deleted the view then set 'NULL' relate data on the db_authorization table
```

File: sql/_35_fig_cake/cbrd_25486/cases/03_drop_javasp.sql [hunk 8]
```diff
@@ -0,0 +1,38 @@
+/* Verified the CBRD-25486
+As-is: if removed the object then remove the related data on the db_authorization table
+To-be: if removed the object then set 'NULL' relate data on the db_authorization table
```
Comment (bagus-kim) [inline sql/_35_fig_cake/cbrd_25486/cases/03_drop_javasp.sql]:
```suggestion
To-be: if deleted the function then set 'NULL' relate data on the db_authorization table
```

File: sql/_35_fig_cake/cbrd_25486/cases/04_drop_object_from_grant_user.sql [hunk 9]
```diff
@@ -0,0 +1,65 @@
+/* Verified the CBRD-25486
+As-is: if removed the object then remove the related data on the db_authorization table
+To-be: if removed the object then set 'NULL' relate data on the db_authorization table
```
Comment (bagus-kim) [inline sql/_35_fig_cake/cbrd_25486/cases/04_drop_object_from_grant_user.sql]:
```suggestion
To-be: Verify that permissions change to NULL even when changing the authorized owner
```

File: sql/_35_fig_cake/cbrd_25486/cases/05_drop_partition_table.sql [hunk 10]
```diff
@@ -0,0 +1,96 @@
+/* Verified the CBRD-25486
+As-is: if removed the object then remove the related data on the db_authorization table
+To-be: if removed the object then set 'NULL' relate data on the db_authorization table
```
Comment (bagus-kim) [inline sql/_35_fig_cake/cbrd_25486/cases/05_drop_partition_table.sql]:
```suggestion
To-be: After granted multiple permissions on a partition table, verify that those permissions change to NULL on owner change
```

**Review by bagus-kim — COMMENTED** [review]
주석이 상세하게 달려 졌으면 분석 시 도움이 되지 않을까 합니다.

File: sql/_35_fig_cake/cbrd_25486/answers/02_drop_view.answer (hunk 3 above)
Comment (swi0110) [reply sql/_35_fig_cake/cbrd_25486/answers/02_drop_view.answer]: 이 grants column의 값이 NULL로 출력되지 않음을 확인하기 위하여, 부득이하게 grants column을 확인하도록 작성했습니다.

**Comment by swi0110** [comment]
@bagus-kim 

우선 리뷰를 남겨주셔서 감사합니다.
code review에 앞서 아래의 내용 참고를 부탁드립니다.

1. test case의 path를 변경하면서 아래와 같이 file name을 변경하였습니다.
관련된 다른 test case와 동일한 위치로 옮기는 과정에서 부득이하게 file name을 변경하였습니다.
혼란을 드려 죄송합니다.
- tc
renamed:    01_drop_table.sql -> cbrd_25486_01.sql
renamed:    02_drop_view.sql -> cbrd_25486_02.sql
renamed:    03_drop_javasp.sql -> cbrd_25486_03.sql
renamed:    04_drop_object_from_grant_user.sql -> cbrd_25486_04.sql
renamed:    05_drop_partition_table.sql -> cbrd_25486_05.sql
- answer
renamed:    01_drop_table.answer -> cbrd_25486_01.answer
renamed:    02_drop_view.answer -> cbrd_25486_02.answer
renamed:    03_drop_javasp.answer -> cbrd_25486_03.answer
renamed:    04_drop_object_from_grant_user.answer -> cbrd_25486_04.answer
renamed:    05_drop_partition_table.answer -> cbrd_25486_05.answer

2. 또한 지적해주신 주석에 대한 내용도, 각 *.sql 파일에 test scenario에 대한 주석을 추가하여, 내용을 보완했습니다.
리뷰에 참고 부탁드립니다.

**Review by bagus-kim — APPROVED** [review]

**Review by jongmin-won — APPROVED** [review]
